api/metrics.go

```diff
@@ -14,6 +14,7 @@ const (
 	MetricContractSet      = "contractset"
 	MetricContractSetChurn = "churn"
 	MetricContract         = "contract"
+	MetricPerformance      = "performance"
```

Comment:
I think it's odd we have to add this metric here, and only use it in newly added code. There should be several places (switch cases) that we have to extend to add this missing metric.